meteor/client/lib/editAttribute/edit-attribute-base.tsx

```diff
@@ -0,0 +1,155 @@
+import { MongoCollection } from '../../../lib/collections/lib'
+import * as React from 'react'
+import { withTracker } from '../ReactMeteorData/ReactMeteorData'
+
+export interface IEditAttributeBaseProps {
+	updateOnKey?: boolean
+	attribute?: string
+	collection?: MongoCollection<any>
+	myObject?: any
+	obj?: any
+	options?: any
+	optionsAreNumbers?: boolean
+	className?: string
+	modifiedClassName?: string
+	invalidClassName?: string
+	updateFunction?: (edit: EditAttributeBase, newValue: any) => void
+	overrideDisplayValue?: any
+	label?: string
+	mutateDisplayValue?: (v: any) => any
+	mutateUpdateValue?: (v: any) => any
+	disabled?: boolean
+	storeJsonAsObject?: boolean
+	/** Defaults to string */
+	arrayType?: 'boolean' | 'int' | 'float' | 'string'
+}
+
+interface IEditAttributeBaseState {
+	value: any
+	valueError: boolean
+	editing: boolean
+}
+
+export class EditAttributeBase extends React.Component<IEditAttributeBaseProps, IEditAttributeBaseState> {
+	constructor(props) {
+		super(props)
+
+		this.state = {
+			value: this.getAttribute(),
+			valueError: false,
+			editing: false,
+		}
+
+		this.handleEdit = this.handleEdit.bind(this)
+		this.handleUpdate = this.handleUpdate.bind(this)
+		this.handleDiscard = this.handleDiscard.bind(this)
+	}
+	handleEdit(inputValue: any, storeValue?: any) {
+		this.setState({
+			value: inputValue,
+			editing: true,
+		})
+		if (this.props.updateOnKey) {
+			this.updateValue(storeValue ?? inputValue)
+		}
+	}
+	handleUpdate(inputValue: any, storeValue?: any) {
+		this.handleUpdateButDontSave(inputValue)
+		this.updateValue(storeValue ?? inputValue)
+	}
+	handleUpdateEditing(newValue) {
+		this.handleUpdateButDontSave(newValue, true)
+		this.updateValue(newValue)
+	}
+	handleUpdateButDontSave(newValue, editing = false) {
+		this.setState({
+			value: newValue,
+			editing,
+		})
+	}
+	handleDiscard() {
+		this.setState({
+			value: this.getAttribute(),
+			editing: false,
+		})
+	}
+	deepAttribute(obj0: any, attr0: string | undefined): any {
```

Comment:
Just curious, do you know why the naming is `obj0` and `attr0` here?

Reply:
I don't know why that naming was decided on, I only know that is prevalent in Sofie